rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -3144,6 +3159,9 @@ public void removeNodeToken(String nodeUrl, String token) throws RMException {
 
     @Override
     public void setNodeTokens(String nodeUrl, List<String> tokens) throws RMException {
+        if (tokens.stream().filter(token -> token == null || token.isEmpty()).findFirst().isPresent()) {
```

Comment:
we can use `tokens.stream().anyMatch()` here.

Reply:
yes, I pushed the modification